packages/fxa-auth-server/lib/routes/subscriptions/paypal-notifications.ts

```diff
@@ -78,7 +83,19 @@ export class PayPalNotificationHandler extends PayPalHandler {
     }
   }
 
-  private async handleMpCancel(message: IpnMerchPmtType) {}
+  private async handleMpCancel(message: IpnMerchPmtType) {
+    const billingAgreement = await getPayPalBAByBAId(message.mp_id);
+    if (billingAgreement.status == 'Cancelled') {
+      return;
+    }
+    const accountCustomer = await getAccountCustomerByUid(billingAgreement.uid);
+    this.stripeHelper.removeCustomerPaypalAgreement(
+      accountCustomer.uid,
+      { id: accountCustomer.stripeCustomerId } as Stripe.Customer,
+      billingAgreement.billingAgreementId
+    );
+    // TODO: Send email to user that they must go to sub management and re-auth PayPal
```

Comment:
We have the email handlers so we can do this now, right? We should also check here to see if they still have paypal in use before sending the email.

Reply:
What do you mean by PayPal in use?

Reply:
Lesley has a function that returns the payment provider, it does that by determining if the customer has any active/past_due subscriptions. If that indicates that the user has paypal on file, we need to send the email.

Reply:
I am not understanding why we want to check if the customer's payment provider is PayPal again? `getPayPalBAByBAId` returns the billing agreement with the uid. With that uid we then find the account customer and the Stripe customer id. So does this not prove that the user Stripe customer with payment method set as PayPal? Can you provide an example on why we would want to check the payment provider before sending the email?

Reply:
Because if the user no longer has any subscriptions that need payment, it doesn't matter if they cancel their billing agreement. There's no reason to send them an email saying they'll lose their subscription if they don't put one on file.